### PR TITLE
[feat] added markdown lint

### PR DESCRIPTION
**Pull request by 0xMukesh**
## 👨‍💻 Changes proposed

Added a markdown lint github action

## ✔️ Check List (Check all the applicable boxes) 
- [x] My code follows the code style of this project.
- [x] All new and existing tests passed. [Needs to be once confirmed by @krishguptadev]
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    name: Lint changelog file
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Lint changelog file
+      uses: docker://avtodev/markdown-lint:v1
+      with:
+        args: '../CHANGELOG.md ../README.md ../CODE_OF_CONDUCT.md'
```
Comment (suitupstan) [inline .github/workflows/lint.yml]: why do we need to use a 3rd party action? Check the markdownlint-cli2 npm package

Comment (0xMukesh) [reply .github/workflows/lint.yml]: @krishguptadev I didn't knew about that, I have looked up for `markdown lint` in GitHub marketplace.

Comment (0xMukesh) [reply .github/workflows/lint.yml]: Changed the github action 👍

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: Markdown Lint
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: DavidAnson/markdownlint-cli2-action@v4
+      with:
```
Comment (suitupstan) [inline .github/workflows/lint.yml]: You have to use the npm package `markdownlint-cli2` not the action 😓

**Review by suitupstan — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml (hunk 2 above)
Comment (0xMukesh) [reply .github/workflows/lint.yml]: > 

Sorry idk how to use the npm packages in GitHub actions, if there is any docs regarding this I will to change it 🥲

Comment (suitupstan) [reply .github/workflows/lint.yml]: You can check this one for reference (feel free to ask for any help on [our discord](https://krish.ninja/discord)): https://github.com/LearnWithCommunity/support/blob/main/.github/workflows/format.yml

**Review by suitupstan — APPROVED** [review]

**Pull request merged**